grafana/grafana
# Explore: Correct the functionality of shift-enter shortcut across all uses

**What this PR does / why we need it**:
This is similar issue as we have addressed in ExploreToolbar - https://github.com/grafana/grafana/pull/34297#issuecomment-846928783. When user runs the query with "Run query button" or trough query editor, we want to always run query and reset cache. 

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/36349

**Special notes for your reviewer**:

How to test this: 
1. In explorer write a query for logs, something that returns results over several minutes probably
2. Use the 'shift-enter' keyboard shortcut, the query should execute
3. Use the histogram to select a small portion of the results - verify that the time picker shows a specific time range now
4. Edit the logql query; either to a different set of label selectors, or adding a filter expression of any kind that matches some of the log lines in view.
5. Use 'shift-enter' keyboard shortcut, the query **WILL** execute
